ethcore/src/executive.rs

```diff
@@ -250,6 +256,13 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 		tracer: &mut T,
 		vm_tracer: &mut V
 	) -> evm::Result<U256> where T: Tracer, V: VMTracer {
+		trace!("Executive::call(params={:?}) self.env_info={:?}", params, self.info);
+		if (params.call_type == CallType::StaticCall || self.static_flag) && params.value.value() > 0.into() {
```

Comment:
`params` are for this call, no? in which case, if this call type is `StaticCall` (or `DelegateCall`, for that matter) then it should be fine even if it is a static context (`static_flag == true`)